src/Indicator.vala

```diff
@@ -190,15 +187,10 @@ public class Session.Indicator : Wingpanel.Indicator {
 
     public override void closed () {}
 
-    private void show_shutdown_dialog () {
-        if (shutdown_dialog == null) {
-            shutdown_dialog = new Session.Widgets.EndSessionDialog (Session.Widgets.EndSessionDialogType.RESTART);
-            shutdown_dialog.destroy.connect (() => { shutdown_dialog = null; });
-            shutdown_dialog.set_transient_for (indicator_icon.get_toplevel () as Gtk.Window);
-            shutdown_dialog.show_all ();
-        }
-
-        shutdown_dialog.present ();
+    private void show_dialog (uint type) {
```

Comment:
Weren't you able to do `show_dialog (Widgets.EndSessionDialogType type)` in vala? That way you wouldn't have to typecast :)

Reply:
@Philip-Scott hello! Yeah, I was unsure about this, but it's fixed now.